libmount/src/hook_subdir.c

```diff
@@ -165,17 +165,29 @@ static int tmptgt_cleanup(struct hookset_data *hsd)
 static int do_mount_subdir(
 			struct libmnt_context *cxt,
 			struct hookset_data *hsd,
-			const char *root,
-			const char *target)
+			const char *root)
 {
 	int rc = 0;
 	const char *subdir = hsd->subdir;
+	const char *target;
 
 #ifdef USE_LIBMOUNT_MOUNTFD_SUPPORT
-	struct libmnt_sysapi *api;
+	struct libmnt_sysapi *api = mnt_context_get_sysapi(cxt);
+
+	/* fallback only; necessary when hook_mount.c during preparation
```

Comment:
Yes, I've thought about it, but it will be necessary to make such behavior optional because, in some cases, we do not want to open the tree or need to open it with non-default options. We can create a different function and use it in places where opening the default tree is desired. Maybe we can directly return the tree FD (from the API object) as it's what is usually needed.
```
/*  return API (unmodified) */
mnt_context_get_sysapi() ;

/* return tree FD, if tree not opened then call mnt_context_open_tree() with default options */
mnt_context_get_tree_fd();
```